transactions/account_linking/add_as_child_multisig.cdc

```diff
@@ -1,80 +1,110 @@
     let authAccountCap: Capability<&AuthAccount>
-    let managerRef: &ChildAccount.ChildAccountManager
-    let info: ChildAccount.ChildAccountInfo
-    let childAccountAddress: Address
+    let linkedAccountAddress: Address
 
     prepare(parent: AuthAccount, child: AuthAccount) {
         
-        /* --- Configure parent's ChildAccountManager --- */
+        /** --- Configure Collection & get ref --- */
         //
-        // Get ChildAccountManager Capability, linking if necessary
-        if parent.borrow<&ChildAccount.ChildAccountManager>(from: ChildAccount.ChildAccountManagerStoragePath) == nil {
-            // Save
-            parent.save(<-ChildAccount.createChildAccountManager(), to: ChildAccount.ChildAccountManagerStoragePath)
+        // Check that Collection is saved in storage
+        if parent.type(at: LinkedAccounts.CollectionStoragePath) == nil {
+            parent.save(
+                <-LinkedAccounts.createEmptyCollection(),
+                to: LinkedAccounts.CollectionStoragePath
+            )
+        }
+        // Link the public Capability
+        if !parent.getCapability<
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, MetadataViews.ResolverCollection}
+            >(LinkedAccounts.CollectionPublicPath).check() {
+            parent.unlink(LinkedAccounts.CollectionPublicPath)
+            parent.link<&LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, MetadataViews.ResolverCollection}>(
+                LinkedAccounts.CollectionPublicPath,
+                target: LinkedAccounts.CollectionStoragePath
+            )
         }
-        // Ensure ChildAccountManagerViewer is linked properly
+        // Link the private Capability
         if !parent.getCapability<
-                &ChildAccount.ChildAccountManager{ChildAccount.ChildAccountManagerViewer}
-            >(ChildAccount.ChildAccountManagerPublicPath).check() {
-            // Link
-            parent.unlink(ChildAccount.ChildAccountManagerPublicPath)
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, NonFungibleToken.CollectionPublic, NonFungibleToken.Receiver, NonFungibleToken.Provider, MetadataViews.ResolverCollection}
+            >(LinkedAccounts.CollectionPrivatePath).check() {
+            parent.unlink(LinkedAccounts.CollectionPrivatePath)
             parent.link<
-                &ChildAccount.ChildAccountManager{ChildAccount.ChildAccountManagerViewer}
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, NonFungibleToken.CollectionPublic, NonFungibleToken.Receiver, NonFungibleToken.Provider, MetadataViews.ResolverCollection}
             >(
-                ChildAccount.ChildAccountManagerPublicPath,
-                target: ChildAccount.ChildAccountManagerStoragePath
+                LinkedAccounts.CollectionPrivatePath,
+                target: LinkedAccounts.CollectionStoragePath
             )
         }
```

Comment:
lot of copied code from the other transaction. It would be nice to have a utility function for this in the contract, but i know we say that passing an AuthAccount into a method is bad. stuff like this makes me maybe want to reconsider that

Reply:
Agreed, transactions like this lead to a ton of duplicate code